wrangler/wrangler.toml.dist

```diff
@@ -1,18 +1,31 @@
-name = "${WRANGLER_WORKER_NAME}"
 type = "javascript"
-usage_model = "bundled"
-webpack_config = "webpack.config.js"
 account_id = "${WRANGLER_ACCOUNT_ID}"
+usage_model = "bundled"
 workers_dev = true
+name = "${WRANGLER_WORKER_NAME}"
 kv_namespaces = [
-   { binding = "META", id = "${WRANGLER_META}" },
-   { binding = "TX_FUNCTIONS", id = "${WRANGLER_TX_FUNCTIONS}" },
-   { binding = "TX_FEES", id = "${WRANGLER_TX_FEES}" },
+  { binding = "META", id = "${WRANGLER_META}" },
+  { binding = "TX_FUNCTIONS", id = "${WRANGLER_TX_FUNCTIONS}" },
 ]
-vars = { STELLAR_NETWORK = "${STELLAR_NETWORK}", HORIZON_URL = "${HORIZON_URL}", TURRET_ADDRESS = "${TURRET_ADDRESS}", TURRET_RUN_URL = "${TURRET_FUNCTION_RUNNER_URL}", XLM_FEE_MIN = ${WRANGLER_XLM_FEE_MIN}, XLM_FEE_MAX = ${WRANGLER_XLM_FEE_MAX}, UPLOAD_DIVISOR = ${WRANGLER_UPLOAD_DIVISOR}, RUN_DIVISOR = ${WRANGLER_RUN_DIVISOR} }
+
+[vars]
+STELLAR_NETWORK = "${STELLAR_NETWORK}"
+HORIZON_URL = "${HORIZON_URL}"
+TURRET_ADDRESS = "${TURRET_ADDRESS}"
+TURRET_RUN_URL = "${TURRET_FUNCTION_RUNNER_URL}"
+XLM_FEE_MIN = ${WRANGLER_XLM_FEE_MIN}
+XLM_FEE_MAX = ${WRANGLER_XLM_FEE_MAX}
+UPLOAD_DIVISOR = ${WRANGLER_UPLOAD_DIVISOR}
+RUN_DIVISOR = ${WRANGLER_RUN_DIVISOR}
 
 [build]
-command = "npm i && npm run build"
+command = "npm run build"
+
 [build.upload]
 format = "modules"
-main = "./shim.mjs"
\ No newline at end of file
+main = "./shim.mjs"
+
+[durable_objects]
+bindings = [
+  { name = "TX_FEES", class_name = "TxFees" }
```

Comment:
Ah! There it is!

Reply:
pew pew! Assuming this resolves all above comments?

Reply:
Another question that popped into my head...

How is a Durable Object created? Does it just call the

```
  constructor(state, env) {
    this.storage = state.storage
  }
```

And provide the relevant params by default? Further, I assume this construction only happens during the initial deploy?

Due to this... is there any realistic way to "upgrade" a durable object? Say, if you needed to make changes to the `TxFees` class? Or will all old fee information get lost unless you manually did something about it?

Reply:
Pasting these here for brevity 
- https://developers.cloudflare.com/workers/learning/using-durable-objects
- https://developers.cloudflare.com/workers/runtime-apis/durable-objects

Reply:
It's "created" when you publish the worker with the `--new-class TxFees` argument. It's "instantiated" as needed whenever it doesn't exist in memory in the Durable Objects ecosystem. The object itself is ephemeral but the storage is permanent, a sort of DO bound KV store.

You can update the script as you please without affecting the underlying storage however if you ever delete the Durable Object it will remove all the storage as well (which is different to Workers vs KV which aren't directly linked). If you needed to move you could either transfer the DO to a new name or just create an export endpoint which could spit out all the underlying storage values for transfer to some other system.

tldr; There's no lock-in fear or inability to change or access underlying data. Also the data is actually permanent outside the script. They are bound to existence of the DO but not the form, so you can change the code as you please and always just call `this.storage` to get at the DO data.

Hopefully that answers the question.

Reply:
Ah, I didn't realize `this.storage` was actually outside the object itself. I assumed it was inside! Cool. Seems the Cloudflare folk thought of that ahead of time!